index.bs

```diff
@@ -6058,7 +6054,7 @@ dictionary MLOperandDescriptor {
   required MLOperandDataType dataType;
 
   // The dimensions field is only required for tensor operands.
-  sequence<unsigned long> dimensions;
+  sequence<unsigned long> dimensions = [];
```

Comment:
We need to follow up its checks against empty array vs against "exists"?

Reply:
FYI: My audit turned up https://github.com/webmachinelearning/webnn/pull/476#issuecomment-1934885013 which have all been tackled. But please look for more!

Reply:
Thank you for the audit Josh. I want to push this through since some of Josh's other changes interact with it, but I'll need to sweep the entire document anyway for coherency, as I've noticed a few changes needed since the transformers change (e.g. the fill sequence `constant` overload is absent from the IDL https://www.w3.org/TR/webnn/#idl-index).